tests/dialects/test_sqlite.py

```diff
@@ -140,6 +140,19 @@ def test_sqlite(self):
             read={"snowflake": "LEAST(x, y, z)"},
             write={"snowflake": "LEAST(x, y, z)"},
         )
+        self.validate_all(
+            "SELECT STR_TO_DATE('2022-01-01', '%Y-%m-%d')",
```

Comment:
Take a look at how `validate_all` is [defined](https://github.com/tobymao/sqlglot/blob/3bd811d67e7001046812f2a8590bd07e23b81c88/tests/dialects/test_dialect.py#L27C14-L27C14): the first argument gets parsed according to the dialect corresponding to the surrounding class (here: SQLite). This means that it's generally SQL which is valid in that dialect, i.e. `STRFTIME` instead of `STR_TO_DATE`.